internal/metadata/request_metadata_parser.go

```diff
@@ -0,0 +1,20 @@
+package metadata
+
+import (
+	"github.com/satsuma-data/node-gateway/internal/jsonrpc"
+)
+
+type RequestMetadataParser struct{}
+
+func (p *RequestMetadataParser) Parse(requestBody jsonrpc.RequestBody) RequestMetadata {
+	result := RequestMetadata{}
+
+	switch requestBody.Method {
+	case "eth_getBalance", "eth_getStorageAt", "eth_getTransactionCount", "eth_getCode", "eth_call", "eth_estimateGas":
```

Comment:
nit: can we include add a link to the Eth docs that say which methods are state methods?